src/chocolatey.tests/infrastructure.app/nuget/ChocolateyNugetCredentialProviderSpecs.cs

```diff
@@ -0,0 +1,320 @@
+                        VisibleToAdminsOnly = false,
+                        EncryptedPassword = NugetEncryptionUtility.EncryptString(Password),
+                        Username = Username,
+                        Key = TargetSourceUrl,
+                        Name = TargetSourceName,
+                        Priority = 1,
+                    },
+                };
+            }
+
+            [OneTimeSetUp]
+            public async Task OneTimeSetup()
+            {
+                await With();
+            }
+
+            public virtual async Task With()
+            {
+                var result = await Provider.GetAsync(TargetSourceUri, proxy: null, CredentialRequestType.Unauthorized, message: string.Empty, isRetry: false, nonInteractive: true, CancellationToken);
+                Result = result.Credentials as NetworkCredential;
+            }
+        }
+
+        public class When_using_explicit_credentials_and_source_param : ChocolateyNugetCredentialProviderSpecsBase
+        {
+            public override void Because()
+            {
+                Configuration.Sources = Configuration.ExplicitSources = TargetSourceUrl;
+                Configuration.SourceCommand.Username = "user";
+                Configuration.SourceCommand.Password = "totally_secure_password!!!";
+            }
+
+            [Fact]
+            public void Creates_a_valid_credential()
+            {
+                Result.Should().NotBeNull();
+            }
+
+            [Fact]
+            public void Provides_the_correct_username()
+            {
+                Result.UserName.Should().Be("user");
+            }
+
+            [Fact]
+            public void Provides_the_correct_password()
+            {
+                Result.Password.Should().Be("totally_secure_password!!!");
+            }
+        }
+
+        public class When_a_source_name_is_provided : ChocolateyNugetCredentialProviderSpecsBase
+        {
+            public override void Because()
+            {
+                Configuration.Sources = TargetSourceUrl;
+                Configuration.ExplicitSources = TargetSourceName;
+            }
+
+            [Fact]
+            public void Finds_the_saved_source_and_returns_the_credential()
+            {
+                Result.Should().NotBeNull();
+            }
+
+            [Fact]
+            public void Provides_the_correct_username()
+            {
+                Result.UserName.Should().Be(Username);
+            }
+
+            [Fact]
+            public void Provides_the_correct_password()
+            {
+                Result.Password.Should().Be(Password);
+            }
+        }
+
+        public class When_a_source_url_matching_a_saved_source_is_provided : ChocolateyNugetCredentialProviderSpecsBase
+        {
+            public override void Because()
+            {
+                Configuration.Sources = Configuration.ExplicitSources = TargetSourceUrl;
+            }

```

Comment:
Again, should be Context.

```suggestion
            public override void Context()
            {
                base.Context();
                Configuration.Sources = Configuration.ExplicitSources = TargetSourceUrl;
            }

```